### ISSUE TITLE
Dataset setup | Split Train:Val:Test

### ISSUE THREAD

**Issue by hdnminh** [MIC-DKFZ/nnUNet#1864]
Hi Team,

I wish you a great last week of the year.

How is dataset setup for training and testing? Have you split data t**rain:test = 80:20**? 
Then you **run** (training process) _5-fold cross-validation_ on the **training** **set**, and **evaluate** (tessting processs) on the **test** **set**?

The **train set** is internally always split into **train** and **val** and the **validation metrics** are also provided during training in the log files?

Thank you in advance.

**Comment by hdnminh** [comment]
Hi @Lars-Kraemer

Can you help me? Thank you so much

**Comment by Lars-Kraemer** [comment]
Hey @hdnminh 

nnUNet uses the following folder structure in the nnUNet_raw datasets:
- imagesTr
- labelsTr
- imagesTs (optional)
- labelsTs (optional)

imagesTr and labelsTr contain your training data, while imagesTs and labelsTs are your test data. How you divide them is up to you. nnUnet does not create a test set for you, instead, all available data (all images/labels in imagesTr/labelsTr) is used for training. As you correctly described, during training, a 5-fold cross-validation is performed on the training set with train:val = 80:20 for each of the 5 folds. After training, a final evaluation is performed on the corresponding validation set (not the test set).
If you want to evaluate your model on the test set, perform the following steps:
1. Train your models 
2. Predict your test data (with single folds or the ensemble) (see `nnUNetv2_predict -h` for more details) 
3. Evaluate your predictions (see `nnUNetv2_evaluate_folder -h` for more details)

I hope this answers your question,

Best,
Lars
